Math/NumberTheory/Primes/Sieve/SmoothSieve.hs

```diff
@@ -0,0 +1,80 @@
+module Math.NumberTheory.Primes.Sieve.SmoothSieve
+  ( smoothData,
+    smoothNumbers,
+  ) where
+
+import Data.Vector (Vector, replicate, findIndices, accumulate, map)
+import qualified Data.Vector as V
+import qualified Data.Vector.Mutable as MV
+import Control.Monad.ST
+import Control.Monad
+-- Finds the number of primes less than or equal to a given number
+import Math.NumberTheory.Primes.Counting.Impl
```

Comment:
It is not a big deal and you certainly can do it, but if there is no specific reason, it is better to stick to publicly exposed modules such as `import Math.NumberTheory.Primes`. This is because the interface of public modules is more or less stable, while internals can be changed by other contributors without restrictions.